src/ServiceClient/Implements/SocketClient/DeribitSocketClient.cs

```diff
@@ -0,0 +1,193 @@
+﻿using Microsoft.Extensions.Options;
+using ServiceClient.Abstractions;
+using ServiceClient.Implements;
+using ServiceClient.Implements.SocketClient.DTOs;
+using System.Net.WebSockets;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace ServiceClient.Implements.SocketClient;
+
+internal class DeribitSocketClient : IDeribitClient
+{
+    private int nextId = 0;
+    private const int readBufferSize = 100;
+    private readonly ClientWebSocket webSocket;
+    private readonly DeribitOptions deribitOptions;
+    public event EventHandler<BookReadedEventArgs>? OnBookReaded;
+    public event EventHandler<TickerReadedEventArgs>? OnTickerReaded;
+
+    public AuthResult? Credentials { get; protected set; }
+    public DeribitSocketClient(IOptions<DeribitOptions> options)
+    {
+        deribitOptions = options.Value;
+        webSocket = new ClientWebSocket()
+        {
+            Options =
+            {
+                KeepAliveInterval = TimeSpan.FromSeconds(deribitOptions.KeepAliveIntervalInSeconds),
+            }
+        };
+    }
+    protected string BookChannel => $"book.{deribitOptions.InstrumentName}.{deribitOptions.BookInterval}";
+    protected string TickerChannel => $"ticker.{deribitOptions.InstrumentName}.{deribitOptions.TickerInterval}";
+
+    private Task SendTestAsync(CancellationToken token)
+    {
+        return SendAsync("public/test", new object(), token);
+    }
+
+    private Task SendAsync<T>(string method, T data, CancellationToken cancellationToken)
+    {
+
+        var request = new Request<T>
+        {
+            JsonRpcVersion = "2.0",
+            Id = ++nextId,
+            Method = method,
+            Parameters = data,
+        };
+        var jsonMessage = JsonSerializer.Serialize(request, jsonOptions);
+        var buffer = new byte[jsonMessage.Length];
+
+        Encoding.Default.GetBytes(jsonMessage, 0, jsonMessage.Length, buffer, 0);
+
+        return webSocket.SendAsync(buffer, WebSocketMessageType.Text, true, cancellationToken);
+    }
+
+    private async Task<string> ReadStringAsync(CancellationToken cancellationToken)
+    {
+        WebSocketReceiveResult response;
+        StringBuilder stringBuilder = new();
+        byte[] readBuffer = new byte[readBufferSize];
+        var receiveValidStates = new[] { WebSocketState.Open, WebSocketState.CloseSent };
+        do
```

Comment:
The three lines above should be declared outside the method as global states to avoid generating garbage on the heap.

Inside the method use:

`stringBuilder.Clear();`

**This method is used in a hot path**

Reply:
I add this to another issue